docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,549 @@
+    |   ||   ||   |    ||   ||   ||        |    |---'    | | ||   ||   ||   ||---'
+    `---'`---'`   '    |`---'`---'`        `---'`---'    ` ' '`---'`   '`---'`---'
+                   `---'
+
+
+Connect the home interface
+--------------------------
+
+Pyfiglet adds some new functionality to FIGlet, such as a feature for installing new
+fonts to the user's ``/home`` directory. But by default, snaps block access to system
+resources like USB devices, the network, and home. *Interfaces* permit access to
+individual resources on the host, be they software, data, or hardware.
+
+To enable writing to the home directory, we must connect the home interface.
+
+Interfaces are established with the ``plug`` key on apps. Let's connect our
+``ukuzama-pyfiglet`` app to the home interface:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+    :emphasize-lines: 4-5
+
+    apps:
+      ukuzama-pyfiglet:
+        command: bin/pyfiglet
+        plugs:
+          - home
+
+If we repack and reinstall the snap, can install a font to our local home directories.
+However, before we repack, let's go back to two keys we skipped at the start.
+
+
+Secure the snap
+~~~~~~~~~~~~~~~
+
+Now that we're handling interfaces and sandboxing -- usually the last step in the
+crafting process -- the ``grade`` and ``confinement`` keys are relevant.
+
+These keys account for the security and stability of the snap. The ``grade`` key is a
+self-attestation of how risky the snap is. When set to ``devmode``, snapd and snap
+stores won't treat it as ready for production. The ``confinement`` key determines
+whether the snap needs special classic confinement, which removes all guardrails and
+grants the snap full system access.
+
+We want our snap to be as safe and secure as possible, so let's change these values to:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    grade: stable
+    confinement: strict
+
+Now, when we build the snap, the snap's access to the host is inverted -- all sensitive
+system resources are blocked unless facilitated by an interface. And when we publish the
+snap, users will be able to install it without the ``--devmode`` flag.
+
+
+Test the interface
+~~~~~~~~~~~~~~~~~~
+
+Now that the snap is confined, we can realistically test the home interface connection.
+
+Build and reinstall the snap, but this time, install it like a production-ready snap:
+
+.. code-block:: bash
+
+    snap install ukuzama-pyfiglet --dangerous
+
+.. note::
+
+    We will continue passing the ``--dangerous`` flag until the snap is published to the
+    Snap Store.
+
+Next, let's gather a font that wasn't included with pyfiglet and install it. Download
+`Small Braille <https://github.com/xero/figlet-fonts/blob/master/smbraille.tlf>`_ from
+the figlet-fonts project. Install it and verify it with:
+
+.. code-block:: bash
+
+    ukuzama-pyfiglet -L ~/Downloads/smbraille.tlf
+    ls ~/.local/pyfiglet/fonts # todo: verify
```

Comment:
Untested and WIP due to project deficiencies.

Reply:
You may run into a problem with `home`, which I don't think allows access to hidden dirs.  Instead, something like:

```yaml
plugs:
  dot-local-pyfiglet:
    interface: personal-files
    read:
      - $HOME/.local/pyfiglet
```

I think that naming convention, `dot-local-pyfiglet`, is loosely enforced by the policy reviewers team when you request auto-connect permissions in the forum.  That may be too much info to note in the tutorial, but I'm mentioning it so you aren't too tempted to change the name.